lib/gst/player/gstplayer.c

```diff
@@ -958,6 +980,7 @@ state_changed_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
       gst_element_query_duration (self->priv->playbin, GST_FORMAT_TIME,
           &duration);
       emit_duration_changed (self, duration);
+      self->priv->is_prerolled = TRUE;
```

Comment:
The idea would be to only create the initial media_info here (for all the streams, etc) and just keep media_info at NULL before that (and if you get all the various signals when media_info==NULL you just ignore them).